test/FastRouteRouterTest.php

```diff
@@ -154,7 +154,7 @@ public function testMatchingRouteShouldReturnSuccessfulRouteResult()
         $result = $router->match($request->reveal());
         $this->assertInstanceOf(RouteResult::class, $result);
         $this->assertTrue($result->isSuccess());
-        $this->assertEquals('/foo', $result->getMatchedRouteName());
+        $this->assertEquals('/foo^GET', $result->getMatchedRouteName());
```

Comment:
A note to anybody who stumbles across this: by fixing the issue, I exposed a problem with this particular test. The assumption was that the route *name* was simply the path. However, that's not the case, nor has it been, for most of the lifetime of this implementation; in order to distinguish between middleware using the same path but different HTTP methods, we had to append the method information auto-magically. Updating the code to return the name picked that up, and hence the change to this assertion.